app/app.go

```diff
@@ -602,10 +602,63 @@ func (app *RegenApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) a
 // EndBlocker application updates every end block
 func (app *RegenApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
 	resp := app.mm.EndBlock(ctx, req)
+
+	const PATCH_HEIGHT = 600
+	lastUpdatedVals := []string{
```

Comment:
We don't have access to tendermint validator set in EndBlocker, so hard coding validator set.